src/arrayMethodSplice.js

```diff
@@ -5,7 +5,78 @@
  */
 function applyCustomSplice() {
   [].__proto__.splice2 = function(start, deleteCount, ...items) {
-    // write code here
+    const deleteCountSplice = deleteCount;
+    const arrayItems = [...items];
+    const argumentsLength = arguments.length;
+    const sourceLength = this.length;
+    const returnedArray = [];
+    let startOfSplice = start;
+    // assign all constants for next work with them
```

Comment:
do not comment code, only in excepted cases, use explicit variable names and clear logic